### PR TITLE
More resources from class generator

### PR DESCRIPTION
**Pull request by myakove**
## Summary by CodeRabbit

- **New Features**
	- Enhanced console output for dry runs with syntax highlighting for better readability.
	- New `ReplicaSet` class for managing Kubernetes ReplicaSets.
	- Introduction of `VirtualMachineInstanceReplicaSet` class for managing virtual machine instance replica sets.
	- Added optional parameters to `Deployment`, `Namespace`, `Project`, and related classes for more flexible configurations.

- **Bug Fixes**
	- Improved error handling for missing required fields in several classes.

- **Deprecations**
	- Deprecated modules for `ReplicaSet`, `VirtualMachineInstanceReplicaSet`, and `VirtualMachineClusterInstancetype`, advising users to transition to new import paths.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: ocp_resources/virtual_machine_instance_replica_set.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from ocp_resources.resource import NamespacedResource, MissingRequiredArgumentError
+
+
+class VirtualMachineInstanceReplicaSet(NamespacedResource):
+    """
+    VirtualMachineInstance is *the* VirtualMachineInstance Definition. It
+    represents a virtual machine in the runtime environment of kubernetes.
+    """
+
+    api_group: str = NamespacedResource.ApiGroup.KUBEVIRT_IO
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/virtual_machine_instance_replica_set.py:13]: This is a class constant, and it is different from class to class

File: ocp_resources/virtual_machine_instance_replica_set.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    def to_dict(self) -> None:
+        super().to_dict()
+
+        if not self.yaml_file:
+            if not all([
+                self.selector,
+                self.template,
+            ]):
+                raise MissingRequiredArgumentError(argument="selector, template")
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/virtual_machine_instance_replica_set.py:70]: We need to check it only if yaml_file is not provided and only if to_dict is called (called only on create())

File: ocp_resources/project.py [hunk 3]
```diff
@@ -1,50 +1,104 @@
...
+    ) -> None:
+        """
+        Args:
+            finalizers(Dict[Any, Any]): Finalizers is an opaque list of values that must be empty to permanently
+              remove object from storage
+
+        """
+        super().__init__(**kwargs)
+
+        self.finalizers = finalizers
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/project.py:48]: finalizers are not mandatory

File: ocp_resources/project.py [hunk 4]
```diff
@@ -1,50 +1,104 @@
...
+
+    def to_dict(self) -> None:
+        super().to_dict()
+
+        if not self.yaml_file:
+            self.res["spec"] = {}
+            _spec = self.res["spec"]
+
+            if self.finalizers:
+                _spec["finalizers"] = self.finalizers
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/project.py:58]: if the user sent us anything we want to set it.

File: ocp_resources/project.py [hunk 5]
```diff
@@ -1,50 +1,104 @@
...
+        Args:
+            description(str): Description is the description to apply to a project
+
+            display_name(str): DisplayName is the display name to apply to a project
+
+        """
+        super().__init__(**kwargs)
+
+        self.description = description
+        self.display_name = display_name
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/project.py:91]: we let the underline API to raise the errors if the user sent wrong values

File: ocp_resources/project.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/project.py:48]: we let the underline API to raise the errors if the user sent the wrong values

**Review by rnetser — APPROVED** [review]

**Pull request merged**